File: src/gestorhospedaje/Hospedaje.java
package gestorhospedaje;
import java.io.Serializable;
import java.util.ArrayList;

public class Hospedaje implements Serializable {
	private static final long serialVersionUID = 1L;
	private String nombre;
	private String locacion;
	private long precio_dia;
	private int estrellas;
	private static ArrayList<Hospedaje> hospedajes = new ArrayList<Hospedaje>();

	//CONSTRUCTORES
	public Hospedaje(String nombre, String locacion, long precio_dia, int estrellas) {
		this.nombre = nombre;
		this.locacion = locacion;
		this.precio_dia = precio_dia;
		this.estrellas = estrellas;
		hospedajes.add(this);
	}
	public int calcularPrecio(int dias) {
		return (int)( dias * this.precio_dia);
	}

	public static ArrayList<Hospedaje> buscarHospedajePorUbicacion (String ubicacion) {
		ArrayList<Hospedaje> hospedajesEnUbicacion = new ArrayList<Hospedaje>();
		for (int i = 0; i < hospedajes.size(); i++)
		{
		  if (hospedajes.get(i).getLocacion().equalsIgnoreCase(ubicacion))
		  {
			  hospedajesEnUbicacion.add(hospedajes.get(i));
		  }
		}
		return hospedajesEnUbicacion;
	}
	
	public static Hospedaje buscarHospedajePorNombre(String nombre) {
		for (int i = 0; i < hospedajes.size(); i++)
		{
		  if (hospedajes.get(i).getNombre().equalsIgnoreCase(nombre))
		  {
			  return hospedajes.get(i);
		  }
		}
		return null;
	}


		public void setLocacion(String locacion) {
			this.locacion = locacion;
		}

		public void setPrecio_dias(long precio_dias) {
			this.precio_dia = precio_dias;
		}

		public static ArrayList<Hospedaje> getHospedajes() {
			return hospedajes;
		}

		public static void setAlojamientos(ArrayList<Hospedaje> hospedajes) {
			Hospedaje.hospedajes = hospedajes;
		}

		public long getPrecio_dia() {
			return precio_dia;
		}

		public void setPrecio_dia(long precio_dia) {
			this.precio_dia = precio_dia;
		}

		public String getLocacion() {
			return locacion;
		}

		public int getEstrellas() {
			return estrellas;
		}

		public void setEstrellas(int estrellas) {
			this.estrellas = estrellas;
		}

		public String getNombre() {
			return nombre;
		}

		public void setNombre(String nombre) {
			this.nombre = nombre;
		}

	}

File: practica02/Practica2VALE/Practica02VALE/SRC/gestorAplicacion/hospedaje/Hospedaje.py
class Hospedaje:
    
    _hospedajes = []

    
    def __init__(self, nombre, locacion, precio_dia, estrellas):
        self._nombre = nombre
        self._locacion = locacion
        self._precio_dia = precio_dia
        self._estrellas = estrellas
        Hospedaje._hospedajes.append(self)



#calcular precio de hospedaje
    def calcularPrecio(self, dias):
        return int((dias * self._precio_dia))

 

    @staticmethod
    def buscarHospedajePorUbicacion(ubicacion):
        hospedajesEnUbicacion = []
        i = 0
        while i < len(Hospedaje._hospedajes):
            if Hospedaje._hospedajes[i].getLocacion().casefold() == ubicacion.casefold():
                hospedajesEnUbicacion.append(Hospedaje._hospedajes[i])
            i += 1
        return hospedajesEnUbicacion

    @staticmethod
    def buscarHospedajePorNombre(nombre):
        i = 0
        while i < len(Hospedaje._hospedajes):
            if Hospedaje._hospedajes[i].getNombre().casefold() == nombre.casefold():
                return Hospedaje._hospedajes[i]
            i += 1
        return None


    def setLocacion(self, locacion):
        self._locacion = locacion

    def setPrecio_dias(self, precio_dias):
        self._precio_dia = precio_dias

    @staticmethod
    def getHospedajes():
        return Hospedaje._hospedajes

    @staticmethod
    def setHospedajes(hospedajes):
        Hospedaje._hospedajes = hospedajes
    def getPrecio_dia(self):
        return self._precio_dia

    def setPrecio_dia(self, precio_dia):
        self._precio_dia = precio_dia

    def getLocacion(self):
        return self._locacion

    def getEstrellas(self):
        return self._estrellas

    def setEstrellas(self, estrellas):
        self._estrellas = estrellas

    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre
